exonum-java-binding/core/src/test/java/com/exonum/binding/core/runtime/ServiceArtifactIdTest.java

```diff
@@ -41,25 +43,66 @@ void parseFromRoundtrip(String serviceId) {
   void valueOf() {
     int runtimeId = 1;
     String name = "test-service";
-    ServiceArtifactId id = ServiceArtifactId.valueOf(runtimeId, name);
+    String version = "0.1";
+    ServiceArtifactId id = ServiceArtifactId.valueOf(runtimeId, name, version);
 
     assertThat(id.getRuntimeId()).isEqualTo(runtimeId);
     assertThat(id.getName()).isEqualTo(name);
+    assertThat(id.getVersion()).isEqualTo(version);
   }
 
   @Test
   void newJavaId() {
     String name = "test-service";
-    ServiceArtifactId id = ServiceArtifactId.newJavaId(name);
+    String version = "test-version";
+    ServiceArtifactId id = ServiceArtifactId.newJavaId(name, version);
 
     assertThat(id.getRuntimeId()).isEqualTo(RuntimeId.JAVA.getId());
     assertThat(id.getName()).isEqualTo(name);
   }
 
+  @Test
+  void fromProto() {
+    ArtifactId artifactId = createArtifactIdMessage();
+
+    ServiceArtifactId serviceArtifactId = ServiceArtifactId.fromProto(artifactId);
+
+    assertThat(serviceArtifactId.getRuntimeId()).isEqualTo(artifactId.getRuntimeId());
+    assertThat(serviceArtifactId.getName()).isEqualTo(artifactId.getName());
+    assertThat(serviceArtifactId.getVersion()).isEqualTo(artifactId.getVersion());
+  }
+
   @Test
   void toStringTest() {
-    ServiceArtifactId id = ServiceArtifactId.valueOf(0, "full-name");
+    ServiceArtifactId id = ServiceArtifactId.valueOf(0, "full-name", "0.1");
 
-    assertThat(id.toString()).isEqualTo("0:full-name");
+    assertThat(id.toString()).isEqualTo("0:full-name:0.1");
   }
+
+  @ParameterizedTest
+  @ValueSource(strings = {
+      "",
+      "1:too-few:components:1.0",
+      "1:com.acme:foo-service:0.1.0:extra-component",
+      " : : ",
+      "1:com acme:foo:1.0",
+      "1 :com.acme:foo:1.0",
+      "1:com.acme:foo service:1.0",
+      "1:com.acme:foo-service:1 0",
+      "1:com.acme:foo-service: 1.0",
+      "1:com.acme:foo-service:1.0 ",
+  })
+  void checkInvalidName(String artifactId) {
+    assertThrows(IllegalArgumentException.class,
+        () -> ServiceArtifactId.parseFrom(artifactId));
+  }
+
+  private ArtifactId createArtifactIdMessage() {
```

Comment:
Make it a constant? 

Protobufs are immutable, this method is not parameterized.